### ISSUE TITLE
Scene editing is slow in complex projects

### ISSUE THREAD

**Issue by KoBeWi** [godotengine/godot#60127]
### Godot version

3.5 beta3

### System information

Windows 10 x64

### Issue description

I recently noticed that the scene editing is much less smooth than it used to be. The difference is so noticeable that I was able to record it.

Here's removing and adding a node in 3.4 beta6
![Godot_v3 4-beta6_win64_EUxHzV4cwV](https://user-images.githubusercontent.com/2223172/162646210-2d2d4ea9-9df5-48c7-8d53-ecfeea520e06.gif)
Literally instant.

And the same operation in 3.5 beta3:
![godot_z60H39d09A](https://user-images.githubusercontent.com/2223172/162645910-86ec91bc-bf2d-415f-a19e-281ec3eca7a1.gif)
The 1 second lag after every action is so visible that it hurts.

I didn't test between which versions exactly this started happening.
Also doesn't seem to happen in 4.0 alpha6.

### Steps to reproduce

1. Have a relatively big scene
2. Do anything that changes scene tree, like reparenting a node or changing name

**Comment by Listwon** [comment]
Will you have some time to test it at least in beta1 and beta2? https://downloads.tuxfamily.org/godotengine/3.5/

**Comment by KoBeWi** [comment]
Same issue in beta1.

**Comment by KoBeWi** [comment]
The issue happens on big scenes. Also I think it might be related to instances. When you have tens of instances, the performance is abysmal. But I have a GUI scene with over a hundred of nodes, without any instance, and there is 0 delay (just like in 3.4).

**Comment by lawnjelly** [comment]
Probably needs an MRP to bisect / profile (unless a contributor knows they have worked on something that might have been a likely suspect).

**Comment by KoBeWi** [comment]
I tried to make an MRP, but obviously (bruh) the issue doesn't in a minimal project. I created a KinematicBody2D scene with a few sprites and an Area2D child, which is similar to what I typically use and no freezes even with a few hundred instances.

Maybe it's related to complexity of the project, like number of class names etc.

**Comment by Calinou** [comment]
You can use a shell script to create a lot of named classes:

https://github.com/godotengine/godot/issues/27333#issuecomment-574670963

**Comment by KoBeWi** [comment]
Still no luck. I created a bunch of classes, with multiple levels of inheritance, and extended one of them in the scene. Then multiplied the number of instances on my test scene and also made it very deep. After that switching scenes became very slow and the editor started lagging, but duplicating nodes was still instant.

So instead I went to my main project and created empty scene with a few instances. The issue appeared with just 10 of them. So I started stripping the instanced scene. It was still causing freezes after removing everything except the root node. So I removed script from the root node and it helped. So then I restored the scene and tinkered with the script. I cleared it completely and removed a few levels of inheritance, leaving just `extends KinematicBody2D` as a sole line. The issue was still noticeable (i.e. duplicating it on the test scene was still slow). So I removed the script and created a new one, exactly the same, and there was no slowdown. So likely something was cached in the script instance (maybe preloads? singleton references?). I'm lost ;_;
